### PR TITLE
Fix test instance creation

### PR DESCRIPTION
**Pull request by dipti-kothari**
Instead of launching one instance at a time, we should
launch all instances together to avoid launching instances
in different subnets
Tested by Oleksandr Zubenko

Signed-off-by: Dipti Kothari <dkothar@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: nccl/common/nccl-common.sh [hunk 1]
```diff
@@ -181,7 +181,7 @@ create_instance() {
                     --query "Instances[*].InstanceId" \
                     --output=text ${addl_args} 2>&1)
             create_instance_exit_code=$?
-            echo ${INSTANCE_IDS}
```
Comment (zhngaj) [inline nccl/common/nccl-common.sh:184]: This was not working?

File: nccl/common/nccl-common.sh [hunk 2]
```diff
@@ -221,31 +221,34 @@ prepare_instance() {
...
-            INSTANCE_STATE="unavailable"
-            while [ ${INSTANCE_STATE} != 'running' ] && [ ${create_instance_attempts} -lt ${create_instance_retries} ] ; do
-                if [ $1 == 'ami_instance' ] ; then
-                    create_instance ${SSHSG} 1 ${prep_ami} ${instance_ami_type}
-                else
-                    create_instance ${SGId} 1 ${AMIS["${AWS_DEFAULT_REGION}"]} ${instance_test_type}
-                fi
-                if [ ${create_instance_exit_code} -ne 0 ]; then
-                    echo "==> Changing the region"
-                    # Start over with new region
```
Comment (zhngaj) [inline nccl/common/nccl-common.sh:235]: In the commit message, we say `...avoid launching instances in different subnets`. I think changing to `region` in commit message is more appropriate.

File: nccl/common/nccl-common.sh (hunk 1 above)
Comment (dipti-kothari) [reply nccl/common/nccl-common.sh:184]: It was working, but we had one instance. Now ${INSTANCE_IDS} will have 3 instance ids separated by spaces, so added quotes

File: nccl/common/nccl-common.sh (hunk 2 above)
Comment (dipti-kothari) [reply nccl/common/nccl-common.sh:235]: We will still launch in a different region if capacity is low in the current region. Previously we were launching one instance at a time, and in the same region switching to different AZ if there was low capacity in a particular AZ.

File: nccl/common/nccl-common.sh (hunk 1 above)
Comment (zhngaj) [reply nccl/common/nccl-common.sh:184]: Thanks, got it.

File: nccl/common/nccl-common.sh (hunk 2 above)
Comment (zhngaj) [reply nccl/common/nccl-common.sh:235]: And now, we ensure that all instances are either launched in same AZ, same region, or failed. Right?

Comment (dipti-kothari) [reply nccl/common/nccl-common.sh:235]: Yes, even if one instance fails to launch we terminate all instances and try again in a different AZ

**Pull request merged**